crates/sui-core/src/authority.rs

```diff
@@ -824,30 +824,40 @@ impl AuthorityState {
         // If commit_certificate returns an error, tx_guard will be dropped and the certificate
         // will be persisted in the log for later recovery.
         let notifier_ticket = self.batch_notifier.ticket(bypass_validator_halt)?;
-        if let Err(err) = self
+        let res = self
             .commit_certificate(
                 inner_temporary_store,
                 certificate,
                 &signed_effects,
                 notifier_ticket,
             )
-            .await
-        {
-            if matches!(err, SuiError::ValidatorHaltedAtEpochEnd) {
-                debug!(
-                    ?digest,
-                    "validator halted and this cert will never be committed"
-                );
-                tx_guard.release();
-            } else {
-                error!(?digest, "commit_certificate failed: {}", err);
+            .await;
+
+        let seq = match res {
+            Err(err) => {
+                if matches!(err, SuiError::ValidatorHaltedAtEpochEnd) {
+                    debug!(
+                        ?digest,
+                        "validator halted and this cert will never be committed"
+                    );
+                    tx_guard.release();
+                } else {
+                    error!(?digest, "commit_certificate failed: {}", err);
+                }
+                return Err(err);
             }
-            return Err(err);
-        }
+            Ok(seq) => seq,
+        };
 
         // commit_certificate finished, the tx is fully committed to the store.
         tx_guard.commit_tx();
 
+        // index certificate
+        let _ = self
+            .post_process_one_tx(seq, &digest)
```

Comment:
we can't do this because validator shares this code path too. It looks like the simplest approach will be to have a queue between this and the post processing job, aka bypassing the batch maker entirely. That is still suboptimal because validator still has a queue that nobody consumes, which however can be avoided by checking if the node is a fullnode (every FN by default enables post processing job)

Reply:
I don't see the early exit, probably because you are playing with the code?